File: frontend/src/interfaces/dropletRepository.ts
// frontend/src/interfaces/dropletRepository.ts

import type { Droplet, CreateDropletRequest, DropletResponse } from '../models/droplet';

export interface DropletRepository {
  createDroplet(request: CreateDropletRequest): Promise<DropletResponse>;
  getDroplet(id: string): Promise<Droplet>;
  listDroplets(): Promise<Droplet[]>;
  deleteDroplet(id: string): Promise<void>;
}

File: backend/core-api/interfaces/dropletRepository.go
// backend/core-api/interfaces/dropletRepository.go

package interfaces

import (
	"github.com/AvinashMahala/ClusterGenie/backend/core-api/models"
)

type DropletRepository interface {
	CreateDroplet(req *models.CreateDropletRequest) (*models.DropletResponse, error)
	GetDroplet(id string) (*models.Droplet, error)
	ListDroplets() ([]*models.Droplet, error)
	DeleteDroplet(id string) error
	UpdateDroplet(d *models.Droplet) error
}
